### PR TITLE
Feature/reader stream refresh tag header

### LINKED ISSUE #4854: Reader: add tag stream header
Use the new `/v1.2/read/tag/:tag/images` endpoint to add a header image to tag streams. See [this Calypso PR](https://github.com/Automattic/wp-calypso/pull/9704) for details, and also [this one](https://github.com/Automattic/wp-calypso/issues/9098).

### PR DESCRIPTION
**Pull request by nbradbury**
Resolves #4854 - uses the new `/v1.2/read/tag/:tag/images` endpoint to display an image header above tag streams. A design review of this PR will be part of #4840, so only the code needs to be reviewed in this PR.

Note: I recommend testing this on a real device - I've seen some rendering issues on emulators that don't appear on real devices.

device-2016-12-01-205213

### CODE REVIEW

File: WordPress/src/main/java/org/wordpress/android/ui/reader/views/ReaderTagHeaderView.java [hunk 1]
```diff
@@ -40,64 +47,81 @@ public ReaderTagHeaderView(Context context, AttributeSet attrs, int defStyleAttr
...
+        txtTagName.setText(tag.getLabel());
+
+        getImageAndAttribution();
     }
 
-    private void toggleFollowStatus() {
-        if (mCurrentTag == null || !NetworkUtils.checkConnection(getContext())) {
-            return;
-        }
+    private void getImageAndAttribution() {
```
Comment (daniloercoli) [inline WordPress/src/main/java/org/wordpress/android/ui/reader/views/ReaderTagHeaderView.java]: I noticed that this method is called twice avery time you change the tag in the Reader. Seems that `ReaderPostAdapter->onBindViewHolder` is executed twice for some reason.

**Comment by daniloercoli** [comment]
Just throwing this here, even if the designer review will be done in another PR :) 
The tag text overlays the picture but it's almost unreadable :)
device-2016-12-02-134911

**Comment by nbradbury** [comment]
> The tag text overlays the picture but it's almost unreadable :)

Fixed in 8b8d17d by adding a translucent gray overlay to the image.

device-2016-12-02-093433

File: WordPress/src/main/java/org/wordpress/android/ui/reader/views/ReaderTagHeaderView.java (hunk 1 above)
Comment (nbradbury) [inline WordPress/src/main/java/org/wordpress/android/ui/reader/views/ReaderTagHeaderView.java]: I wasn't able to get rid of the extra `onBindViewHolder` call - I'm not sure if that's just the way RecyclerView works due to cell resizing, etc. At any rate, I added a cache of header info so the network request doesn't need to be made more than once.

**Comment by daniloercoli** [comment]
:shipit:

**Pull request merged**